<launch>
  <!-- Name to uniquely identify the device -->
  <arg name="camera"           default="primesense" />
  
  <!-- Passthroughs to the openni launch -->
  <arg name="device_id"                      default="#1" />
  <arg name="depth_registration"             default="true" />
  <arg name="color_depth_synchronization"    default="true" />
  <arg name="auto_exposure"                  default="true" />
  <arg name="auto_white_balance"             default="false" />
  
  <!-- Maximum size of image buffer -->
  <arg name="bufsize"               default="100" />
  
  <!-- Whether or not to show images collected every 30 seconds -->
  <arg name="show_framerate"        default="true" />
  
  
  <!-- Launch Openni2 -->
  <include file="$(find openni2_launch)/launch/openni2.launch">
    <arg name="camera"                         value="$(arg camera)" />
    <arg name="device_id"                      value="$(arg device_id)" />
    <arg name="depth_registration"             value="$(arg depth_registration)" />
    <arg name="color_depth_synchronization"    value="$(arg color_depth_synchronization)" />
    <arg name="auto_exposure"                  value="$(arg auto_exposure)" />
    <arg name="auto_white_balance"             value="$(arg auto_white_balance)" />
  </include> 
  
  <node name="stream_image_buffer" pkg="perception" type="image_buffer.py" ns="$(arg camera)/rgb" output="screen" >
    <param name="instream"           value="image_raw" />
    <param name="bufsize"            value="$(arg bufsize)" />
    <param name="show_framerate"     value="$(arg show_framerate)" />
  </node>
  
  <node name="stream_image_buffer" pkg="perception" type="image_buffer.py" ns="$(arg camera)/depth" output="screen" >
    <param name="instream"           value="image_raw" />
    <param name="bufsize"            value="$(arg bufsize)" />
    <param name="show_framerate"     value="$(arg show_framerate)" />
  </node>
  
</launch>
